Competition2017/src/competition/subsystems/drive/commands/DriveToOrientationCommand.java

```diff
@@ -0,0 +1,60 @@
+package competition.subsystems.drive.commands;
+
+import com.google.inject.Inject;
+import xbot.common.command.BaseCommand;
+import xbot.common.math.ContiguousHeading;
+import xbot.common.math.PIDManager;
+import xbot.common.properties.XPropertyManager;
+import competition.subsystems.drive.DriveSubsystem;
+
+public class DriveToOrientationCommand extends BaseCommand{
+
+    DriveSubsystem drive;
+    ContiguousHeading targetHeading;
+    ContiguousHeading currentHeading;
+    
+    public final double defaultPValue = 1/80d;
+    double headingError;
+    
+    private XPropertyManager propMan;
+    private PIDManager headingDrivePid;
+    
+    @Inject
+    public DriveToOrientationCommand(DriveSubsystem driveSubsystem, XPropertyManager propMan) {
+
+        this.drive = driveSubsystem;
+        this.propMan = propMan;
+        
+        headingDrivePid = new PIDManager("HeadingModule", propMan, defaultPValue, 0, 0);
+    }
+
+    public void setHeadings(double heading) {
+        
+        targetHeading = new ContiguousHeading(heading);
+    }
+
+    public void setPIDPropertyName(String propertyName){
+
+        headingDrivePid = new PIDManager(propertyName, propMan, defaultPValue, 0, 0);
+    }
+    
+    public void reset(){
+        
+        headingDrivePid.reset();
+    }
+    
+    @Override
+    public void initialize() {
+        
+    }
+
+    @Override
+    public void execute() {
+        
+        double errorInDegrees = drive.imu.getYaw().difference(targetHeading);
+        double normalizedError = errorInDegrees/180;
```

Comment:
This was copied from my old code - and I can't remember why I did this. It's probably actually easier to guess at initial PID values without the normalization, since (thanks to geometry classes) we have a better grasp of what 30 degrees is rather than "fractions of a half-circle" which this normalization does.

Long story short - we should remove normalization